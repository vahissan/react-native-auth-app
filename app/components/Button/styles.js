import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  container: {
    borderRadius: 2
  },
  button: {
    borderRadius: 2
  }
});

export default styles;