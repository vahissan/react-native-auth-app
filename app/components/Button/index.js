import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-native-elements';
import Theme from "../../theme";
import styles from './styles';

export default class ButtonComponent extends PureComponent {
  static propTypes = {
    title: PropTypes.string.isRequired
  }

  render() {
    return (
      <Button 
        {...this.props}
        containerViewStyle={styles.button}
        buttonStyle={styles.button}
        backgroundColor={Theme.colors.primaryColor}
        raised={true}
      />
    );
  }
}


