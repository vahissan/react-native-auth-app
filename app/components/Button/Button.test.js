import React from 'react';
import { shallow } from 'enzyme';
import Button from './index';

describe('Button component', () => {
  const onPress = jest.fn();
  const title = "Button";
  const wrapper = shallow(<Button title={title} onPress={() => onPress()} />);
  
  it('Should render properly', () => {
    expect(wrapper).toHaveLength(1);
  });

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should call onPress handler when pressed', () => {
    wrapper.simulate('press');
    expect(onPress).toHaveBeenCalled();
  });
});