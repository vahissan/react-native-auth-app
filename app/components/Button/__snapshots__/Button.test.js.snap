// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonComponent
    onPress={[Function]}
    title="Button"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "backgroundColor": "#6b52ae",
      "buttonStyle": Object {
        "borderRadius": 2,
      },
      "containerViewStyle": Object {
        "borderRadius": 2,
      },
      "onPress": [Function],
      "raised": true,
      "title": "Button",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "backgroundColor": "#6b52ae",
        "buttonStyle": Object {
          "borderRadius": 2,
        },
        "containerViewStyle": Object {
          "borderRadius": 2,
        },
        "onPress": [Function],
        "raised": true,
        "title": "Button",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
